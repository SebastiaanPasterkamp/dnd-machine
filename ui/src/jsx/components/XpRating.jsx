import React from 'react';
import PropTypes from 'prop-types';

import LazyComponent from '../components/LazyComponent.jsx';

class XpRating extends LazyComponent
{
    render() {
        const { xpRating } = this.props;
        const value = Math.round(xpRating);

        return <div className="xp-rating inline">
            {value}&nbsp;XP
        </div>;
    }
}

XpRating.propTypes = {
    xpRating: PropTypes.number.isRequired,
}

export default XpRating;