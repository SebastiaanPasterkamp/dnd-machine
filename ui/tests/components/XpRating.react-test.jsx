import React from 'react';
import XpRating from 'components/XpRating.jsx';
import renderer from 'react-test-renderer';

describe('Component: XpRating', () => {
    it('should render without props', () => {
        const tree = renderer.create(
            <XpRating
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render "0 XP"', () => {
        const tree = renderer.create(
            <XpRating
                xpRating={0}
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render a rounded XP number', () => {
        const tree = renderer.create(
            <XpRating
                xpRating={13.37}
                />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
