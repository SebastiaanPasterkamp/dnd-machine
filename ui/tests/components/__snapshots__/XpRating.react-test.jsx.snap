// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: XpRating should render "0 XP" 1`] = `
<div
  className="xp-rating inline"
>
  0
   XP
</div>
`;

exports[`Component: XpRating should render a rounded XP number 1`] = `
<div
  className="xp-rating inline"
>
  13
   XP
</div>
`;

exports[`Component: XpRating should render without props 1`] = `
<div
  className="xp-rating inline"
>
  — XP
</div>
`;
